src/Power.cpp

```diff
@@ -226,19 +224,24 @@ class AnalogBatteryLevel : public HasBatteryLevel
     {
 
         uint32_t raw = 0;
+        uint8_t raw_c = 0; //raw reading counter
 
 #ifndef BAT_MEASURE_ADC_UNIT // ADC1
-#ifdef ADC_CTRL
+#ifdef ADC_CTRL //enable adc voltage divider when we need to read
         pinMode(ADC_CTRL, OUTPUT);
-        digitalWrite(ADC_CTRL, HIGH);
+        digitalWrite(ADC_CTRL, ADC_CTRL_ENABLED);
         delay(10);
```

Comment:
if there aren't filter caps with the resistor value i've seen  the fet should saturate in less than 3ms